arches_component_lab/src/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor.vue

```diff
@@ -0,0 +1,199 @@
+    };
+    nodeData: {
+        config: {
+            imagesOnly: boolean;
+            maxFiles: number;
+            maxFileSize: number;
+        };
+    };
+}>();
+
+const formFieldRef = useTemplateRef("formFieldRef");
+const currentValues = ref<FileReference[]>();
+
+watch(
+    () => props.initialValue,
+    (newValue) => {
+        if (newValue) {
+            currentValues.value = newValue;
+        } else {
+            currentValues.value = [];
+        }
+    },
+);
+
+onMounted(() => {
+    allowedFileTypes.value = props.nodeData.config.imagesOnly ? "image/*" : "*";
+
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        newFiles: [],
+        deletedFiles: [],
+    };
+
+    if (props.initialValue) {
+        currentValues.value = props.initialValue;
+    }
+});
+
+const currentMax = computed(() => {
+    if (currentValues.value) {
+        return (
+            props.nodeData.config.maxFiles - (currentValues.value.length ?? 0)
+        );
+    } else {
+        return props.nodeData.config.maxFiles;
+    }
+});
+
+function resolver(e: FormFieldResolverOptions) {
+    validate(e);
+}
+
+function select(event: FileUploadSelectEvent) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        newFiles: event.files,
+    };
+}
+
+function remove(event: FileUploadRemoveEvent) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        newFiles: event.files,
+    };
+}
+
+function validate(e: FormFieldResolverOptions) {
+    console.log("validate", e);
+}
+
+function deleteImage(fileId: string) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        deletedFiles: [
+            // @ts-expect-error - This is a bug in the PrimeVue types
+            ...formFieldRef.value!.field.states.value.deletedFiles,
+            fileId,
+        ],
+    };
+    if (currentValues.value) {
+        const fileIndex = currentValues.value
+            .map((image) => image.file_id)
+            .indexOf(fileId);
+
+        if (fileIndex != -1) {
+            currentValues.value?.splice(fileIndex, 1);
+        }
+    }
+}
+</script>
+
+<template>
+    <FormField
+        ref="formFieldRef"
+        v-slot="$field"
+        :name="props.nodeAlias"
+        :initial-value="props.initialValue"
+        :resolver="resolver"
+    >
+        <div class="uploadedImagesContainer">
+            <div
+                v-for="(image, index) in currentValues"
```

Comment:
I don't think we should use the index as a key:

https://vuejs.org/guide/essentials/list.html#maintaining-state-with-key
https://stackoverflow.com/questions/44531510/why-not-always-use-the-index-as-the-key-in-a-vue-js-for-loop